src/merkle/store.rs

```diff
@@ -81,6 +81,51 @@ impl MerkleStore {
         Ok(self)
     }
 
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the node at `index` rooted on the tree `root`.
+    ///
+    /// # Errors
+    ///
+    /// This will return `NodeNotInStorage` if the element is not present in the store.
+    pub fn get_node(&self, root: Word, index: NodeIndex) -> Result<Word, MerkleError> {
+        let mut hash: RpoDigest = root.into();
+
+        // Check the root is in the storage when called with `NodeIndex::root()`
+        self.nodes
+            .get(&hash)
+            .ok_or(MerkleError::NodeNotInStorage(hash.into(), index))?;
+
+        for bit in index.bit_iterator().rev() {
+            let node = self
+                .nodes
+                .get(&hash)
+                .ok_or(MerkleError::NodeNotInStorage(hash.into(), index))?;
+            hash = if bit { node.right } else { node.left }
+        }
+
+        Ok(hash.into())
+    }
+
+    /// Returns the path for the node at `index` rooted on the tree `root`.
+    ///
+    /// The path starts at the sibling of the target leaf.
+    ///
+    /// # Errors
+    ///
+    /// This will return `NodeNotInStorage` if the element is not present in the store.
+    pub fn get_path(&self, root: Word, mut index: NodeIndex) -> Result<MerklePath, MerkleError> {
+        let mut path = Vec::with_capacity(index.depth().saturating_sub(1) as usize);
+        while index.depth() > 0 {
+            let sibling = index.sibling();
+            index.move_up();
+            let node = self.get_node(root, sibling)?;
+            path.push(node);
+        }
+        Ok(MerklePath::new(path))
+    }
```

Comment:
not for this PR: I think we should never return an empty path from here. So, the first check here should be whether `root` exists in the store (and later we should also check whether `root` is a leaf or not).

Also not for this PR: we should re-write this implementation of this method as right now it would do multiple traversals of the store to build the path.